timemachine/exchange/exchange_mover.py

```diff
@@ -173,68 +174,17 @@ def exchange_move(self, coords, box):
         # If our system is in a clash free state, a deletion move has delta_Us typically
         # on the order of 35kTs. However, if we start in a clashy state, we can no longer
         # guarantee this. So it's safer to disable this micro-optimization for now.
-
-        # if delta_U_insert > 1000:
-        # micro-optimization, skip deletion moves if insertion work is really high
-        # return coords, box
-
         delta_U_delete = -self.U_fn(coords, box, a_idxs, b_idxs)
         delta_U_total = delta_U_delete + delta_U_insert
 
-        if delta_U_total < -1000:
-            np.savez(
-                "debug_overflow_" + str(self.t_count) + ".npz",
-                trial_coords=trial_coords,
-                nb_params=self.nb_params,
-                coords=coords,
-                a_idxs=a_idxs,
-                b_idxs=b_idxs,
-            )
-            print("MC overflow detected", delta_U_delete, delta_U_insert)
-
         # convert to inf if we get a nan
         if np.isnan(delta_U_total):
             delta_U_total = np.inf
```

Comment:
Side note: this conversion is no longer necessary, since the implementation will never accept a proposal whose `log_acceptance_probability` is nan
https://github.com/proteneer/timemachine/blob/68040c238553a210164cdb1dc01d0cb71cd394a1/timemachine/md/moves.py#L32-L34.

i.e.
```python
alpha < np.exp(np.nan)
```
will always evaluate to `False` for any realizable value of `alpha`.

Side-side note: Maybe we want to log / warn when `log_acceptance_probability` is nan?

Reply:
Ah good point

Reply:
Note that this does require us to propagate the delta_U `nan` into a log_acceptance_probability `nan`, i.e. I think this can only be removed after we change `min` to `np.min` below?